File: updater/src/main/java/updater/identifiers/SecureRandomCallable.java
package updater.identifiers;

import updater.utility.AbstractIdentifier;
import updater.utility.ClassWrapper;

public class SecureRandomCallable extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.getInterfaces().size() == 1 &&
                classNode.getInterfaces().get(0).equals("java/util/concurrent/Callable") &&
                classNode.getInstanceFields().isEmpty();
    }
}

File: updater-runestar/src/main/java/org/runestar/client/updater/mapper/identifiers/disabled/SecureRandomCallable.kt
package org.runestar.client.updater.mapper.identifiers.disabled

import org.runestar.client.updater.mapper.abstractclasses.IdentityMapper
import org.runestar.client.updater.mapper.predicateutilities.predicateOf
import org.runestar.client.updater.mapper.predicateutilities.type
import org.runestar.client.updater.mapper.wrappers.Class2
import org.runestar.client.updater.mapper.predicateutilities.and
import java.util.concurrent.Callable

class SecureRandomCallable : IdentityMapper.Class() {

    override val predicate = predicateOf<Class2> { it.interfaces == listOf(Callable::class.type) }
            .and { it.instanceFields.isEmpty() }
}